hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1240,14 +1240,19 @@ int MDMParser::_cbCFUN(int type, const char *buf, int len, int* i) {
     return WAIT;
 }
 
-int MDMParser::_cbUBANDMASK(int type, const char *buf, int len, uint32_t* i)
+int MDMParser::_cbUBANDMASK(int type, const char *buf, int len, uint64_t* i)
 {
-    if ((type == TYPE_PLUS) && i){
+    if ((type == TYPE_PLUS) && i) {
         uint64_t a;
+        char s[24] = {};
+        char* pEnd = &s[0];
         *i = 0;
         // \r\n+UBANDMASK: 0,6170,1,524420\r\n
-        if (sscanf(buf, "\r\n+UBANDMASK: 0,%llu", &a) == 1) {
-            *i = (uint32_t)(a & 0xffffffff); // limit to 32-bit (Cat-M R410-02B supports up to band 28)
+        if (sscanf(buf, "\r\n+UBANDMASK: 0,%23[^,\n]", s) == 1) {
```

Comment:
I'm reading the R4 AT reference and it seems that `AT+UBANDMASK?` may give multiple strings. Should we guard for that?

![image](https://user-images.githubusercontent.com/388478/92642948-2d64ec00-f30b-11ea-8323-aca1f19e8663.png)

Reply:
It does currently spit out 0,bandsm1,1,bandnb1.  But the scanf only matches what it matches and the rest is ignored.  We don't want to match NB1 or GSM, so it should be ok as-is.